crates/sui-core/src/authority_aggregator.rs

```diff
@@ -517,6 +524,47 @@ where
         Ok(accumulated_state)
     }
 
+    /// Like quorum_map_then_reduce_with_timeout, but for things that need only a single
+    /// successful response, such as fetching a Transaction from some authority.
+    /// This is intended for cases in which byzantine authorities can time out or slow-loris, but
+    /// can't give a false answer, because e.g. the digest of the response is known, or a
+    /// quorum-signed object such as a checkpoint has been requested.
+    pub(crate) async fn quorum_once_with_timeout<'a, S, FMap>(
+        &'a self,
+        // The async function used to apply to each authority. It takes an authority name,
+        // and authority client parameter and returns a Result<V>.
+        map_each_authority: FMap,
+        timeout_each_authority: Duration,
+    ) -> Result<S, SuiError>
+    where
+        FMap: Fn(AuthorityName, &'a SafeClient<A>) -> AsyncResult<'a, S, SuiError>,
+    {
+        let authorities_shuffled = self.committee.shuffle_by_stake();
```

Comment:
On nodes, no, and this mostly won't be used on validators.

However, I will be adding a similar feature to this soon, since I want to use this in `get_one_checkpoint` (which currently does sampling by hand) which requires restriction to a set of available authorities.